eth/transactionManager.go

```diff
@@ -0,0 +1,241 @@
+	// used by many clients: https://github.com/ethereum/go-ethereum/blob/01a7e267dc6d7bbef94882542bbd01bd712f5548/core/tx_pool.go#L148
+	// We add a little extra in addition to the 10% price bump just to be sure
+	gasPrice := new(big.Int).Add(
+		new(big.Int).Add(
+			tx.GasPrice(),
+			new(big.Int).Div(
+				tx.GasPrice(),
+				big.NewInt(10),
+			),
+		),
+		big.NewInt(10),
+	)
+
+	suggestedGasPrice := tm.gpm.GasPrice()
+
+	// If the suggested gas price is higher than the bumped gas price, use the suggested gas price
+	// This is to account for any wild market gas price increases between the time of the original tx submission and time
+	// of replacement tx submission
+	// Note: If the suggested gas price is lower than the bumped gas price because market gas prices have dropped
+	// since the time of the original tx submission we cannot use the lower suggested gas price and we still need to use
+	// the bumped gas price in order to properly replace a still pending tx
+	if suggestedGasPrice.Cmp(gasPrice) == 1 {
+		gasPrice = suggestedGasPrice
+	}
+
+	// Bump gas price exceeds max gas price, return early
+	max := tm.gpm.MaxGasPrice()
+	if gasPrice.Cmp(max) > 0 {
+		return nil, fmt.Errorf("Replacement gas price exceeds max gas price suggested=%v max=%v", gasPrice, max)
+	}
+
+	// Replacement raw tx uses same fields as old tx (reusing the same nonce is crucial) except the gas price is updated
+	newRawTx := types.NewTransaction(tx.Nonce(), *tx.To(), tx.Value(), tx.Gas(), gasPrice, tx.Data())
+
+	newSignedTx, err := tm.am.SignTx(newRawTx)
+	if err != nil {
+		return nil, err
+	}
+
+	sendErr := tm.eth.SendTransaction(context.Background(), newSignedTx)
+	txLog, err := newTxLog(tx)
+	if err != nil {
+		txLog.method = "unknown"
+	}
+	if sendErr == nil {
+		glog.Infof("\n%vEth Transaction%v\n\nReplacement transaction: \"%v\".  Gas Price: %v \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, newSignedTx.GasPrice().String(), err, strings.Repeat("*", 75))
+	} else {
+		glog.Infof("\n%vEth Transaction%v\n\nReplacement transaction: \"%v\".  Hash: \"%v\".  Gas Price: %v \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, newSignedTx.Hash().String(), newSignedTx.GasPrice().String(), strings.Repeat("*", 75))
+	}
+
+	return newSignedTx, sendErr
+}
+
+func (tm *TransactionManager) checkTxLoop() {
+	for {
+		tm.cond.L.Lock()
+		for len(tm.queue) == 0 {
+			tm.cond.Wait()
+
+			select {
+			case <-tm.quit:
+				glog.V(common.DEBUG).Info("Stopping transaction manager")
+				return
+			default:
+			}
+		}
+
+		tx := tm.queue.pop()
+		tm.cond.L.Unlock()
+
+		var (
+			receipt *types.Receipt
+			err     error
+		)
+
+		for i := 0; i <= tm.maxReplaceMents; i++ {
+			if receipt, err = tm.wait(tx); err == context.DeadlineExceeded && tm.maxReplaceMents > 0 {
+				newTx, replaceErr := tm.replace(tx)
+				if replaceErr != nil {
+					glog.Error(replaceErr)
```

Comment:
I think `replaceError` should be propagated back to the feed so that a consumer of the feed can get error information about the failed replacement. At the moment, after we break here, the value sent to the feed will just include the `err` variable which would just be `context.DeadlineExceeded`.